aspnetcore/blazor/data-binding.md

```diff
@@ -0,0 +1,339 @@
+---
+title: ASP.NET Core Blazor data binding
+author: guardrex
+description: Learn about data binding scenarios for components and DOM elements in Blazor apps.
+monikerRange: '>= aspnetcore-3.1'
+ms.author: riande
+ms.custom: mvc
+ms.date: 02/12/2020
+no-loc: [Blazor, SignalR]
+uid: blazor/data-binding
+---
+# ASP.NET Core Blazor data binding
+
+By [Luke Latham](https://github.com/guardrex) and [Daniel Roth](https://github.com/danroth27)
+
+Data binding to both components and DOM elements is accomplished with the [`@bind`](xref:mvc/views/razor#bind) attribute. The following example binds a `CurrentValue` property to the text box's value:
+
+```razor
+<input @bind="CurrentValue" />
+
+@code {
+    private string CurrentValue { get; set; }
+}
+```
+
+When the text box loses focus, the property's value is updated.
+
+The text box is updated in the UI only when the component is rendered, not in response to changing the property's value. Since components render themselves after event handler code executes, property updates are *usually* reflected in the UI immediately after an event handler is triggered.
+
+Using `@bind` with the `CurrentValue` property (`<input @bind="CurrentValue" />`) is essentially equivalent to the following:
+
+```razor
+<input value="@CurrentValue"
+    @onchange="@((ChangeEventArgs __e) => CurrentValue = 
+        __e.Value.ToString())" />
+        
+@code {
+    private string CurrentValue { get; set; }
+}
+```
+
+When the component is rendered, the `value` of the input element comes from the `CurrentValue` property. When the user types in the text box and changes element focus, the `onchange` event is fired and the `CurrentValue` property is set to the changed value. In reality, the code generation is more complex because `@bind` handles cases where type conversions are performed. In principle, `@bind` associates the current value of an expression with a `value` attribute and handles changes using the registered handler.
+
+In addition to handling `onchange` events with `@bind` syntax, a property or field can be bound using other events by specifying an [`@bind-value`](xref:mvc/views/razor#bind) attribute with an `event` parameter ([`@bind-value:event`](xref:mvc/views/razor#bind)). The following example binds the `CurrentValue` property for the `oninput` event:
```

Comment:
This was a change asked for by @Rick-Anderson ... it was made across the repo.

Reply:
https://github.com/aspnet/AspNetCore.Docs/pull/16042#discussion_r354579403

... and there's a PR with work to make the change everywhere.

Reply:
@scottaddie Here's the commit that did it ...

https://github.com/aspnet/AspNetCore.Docs/pull/16042/commits/3e3a335ec6605f7b9a9733666c18e650151fc003

Do you cats want to confer on it and possibly revert? I can revert all of them here if you guys make the call. Otherwise, we can go forward as they are all consistently code fenced in links, including attributes ... e.g. ...

```
[`[JSInvokable]`](xref:blazor/javascript-interop#invoke-net-methods-from-javascript-functions)
```

Whatever we do, let's try to resolve this now ...... gotta get 🏃 on my next task.

Reply:
I vote for link text without back ticks, but it's not worth holding up merging of the PR.

Reply:
Very well ..... I merely *prefer* unstyled link text ... no vote ... I'm not on the docs team. lol